client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -177,6 +177,15 @@
      */
     ReaderSegmentDistribution getReaderSegmentDistribution();
 
+    /**
+     * Publishes the given stream-cut to the controller in order to truncate the given stream if
+     * the reader group is a subscriber
+     *
+     * @param stream the stream to be truncated.
+     * @param streamCut the stream-cut with which truncation should be done.
+     */
+    void updateTruncationCheckpoint(Stream stream, StreamCut streamCut);
```

Comment:
also with this api users are indicating that they have consumed the data till a particular streamcut.. maybe we should rename this api. it is not clear what truncation checkpoint is and it may not be very helpful for users to think that the stream will be truncated at this checkpoint after they invoke this api, which is not the case.

Reply:
I've renamed the api according to the [PDP](https://github.com/pravega/pravega/wiki/PDP-47:-Pravega-Streams:-Consumption-Based-Retention#updating-reader-group-configuration)